plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -159,14 +167,15 @@ func findNodesThatFit(
 		// Create filtered list with enough space to avoid growing it
 		// and allow assigning.
 		filtered = make([]*api.Node, len(nodes))
-		meta := predicates.PredicateMetadata(pod, nodeNameToInfo)
+
 		errs := []error{}
 
 		var predicateResultLock sync.Mutex
 		var filteredLen int32
 		checkNode := func(i int) {
 			nodeName := nodes[i].Name
-			fits, failedPredicates, err := podFitsOnNode(pod, meta, nodeNameToInfo[nodeName], predicateFuncs)
+			// pFactory takes a pod, node info, and creates a black-box type which can be used by the predicate.
+			fits, failedPredicates, err := podFitsOnNode(pod, pFactory(pod, nodeNameToInfo), nodeNameToInfo[nodeName], predicateFuncs)
```

Comment:
Can can't call pFactory() for every node separately. You have to call it once at the beginining (where it used to be before).